File: demos/retrofit-demo-java/app/src/main/java/com/smd/retrofitdemo/PeopleInSpaceViewModel.java
package com.smd.retrofitdemo;

import android.app.Application;
import com.smd.retrofitdemo.network.PeopleInSpace;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

/**
 * ViewModel for obtaining data about the current astronauts in space.
 * It retrieves the data from the PeopleInSpaceRepository.
 */
public class PeopleInSpaceViewModel extends AndroidViewModel {

    private LiveData<PeopleInSpace> peopleInSpaceLiveData;
    private final PeopleInSpaceRepository peopleInSpaceRepository;

    public PeopleInSpaceViewModel(Application application) {
        super(application);
        peopleInSpaceRepository = new PeopleInSpaceRepository();
    }

    public void init() {
        peopleInSpaceLiveData = peopleInSpaceRepository.getPeopleInSpaceResponseLiveData();
    }

    public LiveData<PeopleInSpace> getPeopleInSpaceLiveData() {
        return peopleInSpaceLiveData;
    }

    public void getPeopleInSpace() {
        peopleInSpaceRepository.getPeopleInSpace();
    }
}

File: lab5-demos/retrofit-demo-kotlin/app/src/main/java/com/smd/retrofitdemo/PeopleInSpaceViewModel.kt
package com.smd.retrofitdemo

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import com.smd.retrofitdemo.network.PeopleInSpace

/**
 * ViewModel for obtaining data about the current astronauts in space.
 * It retrieves the data from the PeopleInSpaceRepository.
 */
class PeopleInSpaceViewModel(application: Application) : AndroidViewModel(
    application
) {
    var peopleInSpaceLiveData: LiveData<PeopleInSpace?>? = null
        private set
    private val peopleInSpaceRepository: PeopleInSpaceRepository = PeopleInSpaceRepository()
    fun init() {
        peopleInSpaceLiveData = peopleInSpaceRepository.peopleInSpaceResponseLiveData
    }

    val peopleInSpace: Unit
        get() {
            peopleInSpaceRepository.peopleInSpace
        }
}